bundles/org.eclipse.jface/src/org/eclipse/jface/resource/URLImageDescriptor.java

```diff
@@ -59,6 +61,16 @@ public URLImageFileNameProvider(String url) {
 		public String getImagePath(int zoom) {
 			URL tempURL = getURL(url);
 			if (tempURL != null) {
+				SVGRasterizer rasterizer = SVGRasterizerRegistry.getRasterizer();
+				if (rasterizer != null) {
+					try (InputStream in = getStream(tempURL)) {
+						if (rasterizer.isSVGFile(in)) {
```

Comment:
If we would just check the file-extension here we would save creating an `InputStream` for each image to render. The latter is probably a significant performance overhead.
Furthermore it would save us from making `SVGRasterizerRegistry` and `SVGRasterizer` public.

Reply:
Please see my [comment](https://github.com/eclipse-platform/eclipse.platform.swt/pull/1638/commits/44e6f19d6eaa1cf09e59e1970b3e6c9c52a0285b#r1881066057).

We could nevertheless change the isSVGFile() call in Platform UI to a file-extension check. This way the method can't be deleted but `SVGRasterizerRegistry` and `SVGRasterizer` could be internal.

Reply:
I made `SVGRasterizerRegistry` internal and changed isSVGFile to a file-extension check.